packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -155,6 +155,80 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform  {
     return _projectInfo ??= await xcodeProjectInterpreter.getInfo(hostAppRoot.path);
   }
   XcodeProjectInfo? _projectInfo;
+
+  /// The build settings for the host app of this project, as a detached map.
+  ///
+  /// Returns null, if Xcode tooling is unavailable.
+  Future<Map<String, String>?> buildSettingsForBuildInfo(
+      BuildInfo? buildInfo, {
+        String? scheme,
+        String? configuration,
+        String? target,
+        EnvironmentType environmentType = EnvironmentType.physical,
+        String? deviceId,
+        bool isWatch = false,
+      }) async {
```

Comment:
Is this formatting right? I believe named args shouldn't be indented in this repo's style. Example: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/text.dart

```suggestion
  Future<Map<String, String>?> buildSettingsForBuildInfo(
    BuildInfo? buildInfo, {
    String? scheme,
    String? configuration,
    String? target,
    EnvironmentType environmentType = EnvironmentType.physical,
    String? deviceId,
    bool isWatch = false,
  }) async {
```